crates/sequencer/src/sequencer.rs

```diff
@@ -611,13 +612,35 @@ where
 
     #[instrument(level = "trace", skip(self), err, ret)]
     pub async fn resubmit_pending_commitments(&mut self) -> anyhow::Result<()> {
-        let pending_commitments = self.ledger_db.get_pending_commitments_l2_range()?;
-        debug!("Pending commitments: {:?}", pending_commitments);
-        for (l2_start, l2_end) in pending_commitments {
-            let commitment_info = commitment_controller::CommitmentInfo {
-                l2_height_range: l2_start..=l2_end,
-            };
-            self.submit_commitment(commitment_info, true).await?;
+        let pending_db_commitments = self.ledger_db.get_pending_commitments_l2_range()?;
+        debug!("Pending db commitments: {:?}", pending_db_commitments);
+        let pending_da_commitments = self.get_pending_da_commitments().await;
+        debug!("Pending da commitments: {:?}", pending_da_commitments);
+        // TODO: also take mined DA blocks into account
+        for (l2_start, l2_end) in pending_db_commitments {
+            if pending_da_commitments.iter().any(|commitment| {
+                commitment.l2_start_block_number == l2_start.0
+                    && commitment.l2_end_block_number == l2_end.0
+            }) {
+                // Update last sequencer commitment l2 height
+                match self.ledger_db.get_last_sequencer_commitment_l2_height()? {
```

Comment:
I'd get last sequencer commitment l2 height before the for loop then compare to that variable and set if needed

```
last = get_last_commitment()

for commitment in pending():
  if !commitment.in_mempool():
      resubmit()
  else:
      delete commitment
   
  if commitment >= last:
      set_last(commitment)
      last = commitment
         

```

Reply:
makes 1000% sense. done

Reply:
now i think, makes 10% sense. because because in the else branch submit_commitment could also potentially update last_commitment_l2_height. we could also track and retrieve updates after submit_commitment finishes, but imo its better as it is right now, especially considering that it will almost always go to else { submit_commitmen } branch